src/Material/Menu.elm

```diff
@@ -0,0 +1,523 @@
+                      |> List.head
+                      |> fromJust "Menu.view offsetHeight"
+
+                height =
+                  case model.oracle of
+                    Nothing -> 0
+                    Just oracle -> oracle.menu.bounds.height
+
+                itemDelay =
+
+                  if model.alignment == topLeft ||
+                     model.alignment == topRight then
+
+                      (height - offsetTop n - offsetHeight n) / height * transitionDuration
+                      |> toString
+                      |> flip (++) "s"
+
+                    else
+
+                      ((offsetTop n) / height * transitionDuration)
+                      |> toString
+                      |> (flip (++) "s")
+              in
+
+                    styled li
+                    ( (if item.enabled then
+                          Style.attribute (onClick addr Oracle.decode' (Close' n))
+                        else
+                          Style.attribute (Html.attribute "disabled" "disabled"))
+                      :: cs "mdl-menu__item"
+                      :: css' "transition-delay" itemDelay (isJust model.open)
+                      :: cs' "mdl-js-ripple-effect" model.ripple
+                      :: cs' "mdl-menu__item--full-bleed-divider" item.divider
+                      :: Style.attribute (Html.property "tabindex" (string "-1"))
+                      :: Style.attribute (Ripple.downOn "mousedown" (Signal.forwardTo addr (Ripple n)))
+                      :: Style.attribute (Ripple.downOn "touchstart" (Signal.forwardTo addr (Ripple n)))
+                      :: Style.attribute (Ripple.upOn "mouseup" (Signal.forwardTo addr (Ripple n)))
+                      :: Style.attribute (Ripple.upOn "mouseleave" (Signal.forwardTo addr (Ripple n)))
+                      :: Style.attribute (Ripple.upOn "touchend" (Signal.forwardTo addr (Ripple n)))
+                      :: Style.attribute (Ripple.upOn "blur" (Signal.forwardTo addr (Ripple n)))
+                      :: []
+                    )
+
+                    ( (::) item.html <|
+
+                        if model.ripple then
+
+                            [ Ripple.view
+                              ( Signal.forwardTo addr (Ripple n))
+                              [ Html.class "mdl-menu__item-ripple-container" ]
+                              ( Dict.get n model.items
+                                |> Maybe.withDefault Ripple.model)
+                            ]
+
+                          else
+                            []
+                    )
+
+          in
+            List.map2 makeItem [1..List.length items] items
+        )
+      ]
+  ]
+
+
+-- HELPER
+
+
+onClick : Signal.Address Action -> Decoder Oracle -> (Oracle -> Action) -> Attribute
+onClick addr decoder action =
+  Html.onWithOptions
+  "click"
+  defaultOptions
+  decoder
+  (Signal.message addr << action)
+
+
+isJust : Maybe a -> Bool
```

Comment:
You'd have `isJust x == (x /= Nothing)`. I think it's easier to get rid of `isJust` and simply do `x /= Nothing` whenever you need it.